RTM: Refactor unit tests for rspec-puppet-facts/facterdb

Phil-Friderici opened the pull request:
Rename hash keys to same value as is used in on_supported_os for main class.
Renamed fixture files to also use OS names from facterdb.

Still need to:
- [x] refactor accesslogin_spec.rb and limits_spec.rb
- [x] remove platforms hash from spec_helper_local.rb

Commit:
Refactor unit tests for main class

anders-larsson commented:
Looks good. But some of the data is AFAIK shared between the spec test classes? There are very few though.

Commit:
Move platforms hash to ruby functions

Phil-Friderici commented:
I have decided to move the platform hash to ruby functions, just the way I did with the vmware module. Should be easier to read and maintain.
Next step is to let the ruby functions deliver the same data as hiera does.

Commit:
Fix EL8 fixture file for pam_password_auth_ac

The fixture file for testing had wrong content.
Fix files(os) in spec_platforms to actually test that file on EL8 & Co.

Phil-Friderici commented:
I am still somewhat confused. The fixture for pam_password_auth_ac on EL8 had strange content that didn't match with the created file. I didn't saw that error earlier as I had a wrong list of common files in the ruby functions to mimic hiera data.

I am not sure if that file was actually tested in the past as there are no Travis runs to inspect. Also I am not able to run the old tests in my environment.

@ghoneycutt maybe you could have a deeper look into that ?

Commit:
Align variable names in manifest and unit tests

Commit:
Remove platforms from spec_helper_local.rb

- create unit tests accordingly in init_spec.rb

ghoneycutt commented:
> I am still somewhat confused. The fixture for pam_password_auth_ac on EL8 had strange content that didn't match with the created file. I didn't saw that error earlier as I had a wrong list of common files in the ruby functions to mimic hiera data.
> 
> I am not sure if that file was actually tested in the past as there are no Travis runs to inspect. Also I am not able to run the old tests in my environment.
> 
> @ghoneycutt maybe you could have a deeper look into that ?

It might be easier to just grab a stock EL8 system and code to that than try and figure out the past.

Phil-Friderici commented:
> It might be easier to just grab a stock EL8 system and code to that than try and figure out the past.

I did this and fixed the fix and rebased the commit.

Commit:
Update to PDK 2.4.0

Commit:
Remove unneeded CI files

anders-larsson commented:
Are we satisfied with the state of this and merge it into the other pull request?

BR

Commit:
Add test for 100% resource coverage

Phil-Friderici commented:
I think now with 100% resource coverage we are complete :)
Yes, please merge :)

Phil-Friderici commented on metadata.json:
```diff
@@ -92,12 +92,6 @@
         "15"
       ]
     },
-//    {
-//      "operatingsystem": "OpenSuSE",
-//      "operatingsystemrelease": [
-//        "13.1"
-//      ]
-//    },
```
`pdk update ` did remove this deactive section.
Don't know why it got deactivated in first place.

anders-larsson replied:
When I ran pdk test unit I got errors than nsswitch module doesn't support openSuse. You don't get those?

Phil-Friderici replied:
I didn't try. It shouldn't complain when `manage_nsswitch` is set to false for OpenSuse. But to speak frankly I don't know if that would make the problem even worse ;)

Phil-Friderici replied:
Seems that OpenSuse 13.x is out of support anyway.
https://en.opensuse.org/Lifetime#Discontinued_distributions

anders-larsson replied:
Added a comment about it here https://github.com/ghoneycutt/puppet-module-pam/pull/243/files/16cbb9d3b04e8951920e798eab80fce55b8eb4bb#diff-12e598daecade60b04e493f45c134fec9287ac7471114d95d1995b4fb651a6c8.

Yes I don't know it if should be supported or not.

Phil-Friderici replied:
Maybe we should discuss the OpenSuse topic in PR https://github.com/ghoneycutt/puppet-module-pam/pull/243
